### PR TITLE
move `deploy` and `login` to `spin cloud`

### LINKED ISSUE #1117: move cloud commands under spin cloud
Going forward, it would be nice to move cloud related commands under the spin cloud command.

$ spin login -> spin cloud login
$ spin deploy -> spin cloud deploy

For convenience, we should leave aliases for `spin login` and `spin deploy` but all other cloud related commands should be added under `spin cloud`.

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #1117

### CODE REVIEW

**Review by fibonacci1729 — APPROVED** [review]

File: src/commands.rs [hunk 1]
```diff
@@ -4,12 +4,14 @@
 pub mod bindle;
 /// Commands for building Spin applications.
 pub mod build;
-/// Command for deploying a Spin app to Hippo
+/// Commands for building Spin applications.
```
Comment (rajatjindal) [inline src/commands.rs:7]: nit, we should fix this comment.

File: src/commands/cloud.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+use anyhow::Result;
+
+use super::deploy::DeployCommand;
+use super::login::LoginCommand;
+
+/// Commands for publishing applications to Fermyon Cloud.
+#[derive(Subcommand, Debug)]
+pub enum CloudCommands {
+    /// Package and upload to Fermyon Cloud
+    #[clap(alias = "deploy")]
```
Comment (radu-matei) [inline src/commands/cloud.rs:11]: The alias doesn't seem to work for me, as it seems to expect a plugin:

```
$ spin deploy       
Unknown command: plugin 'deploy' not found at expected location /Users/radu/Library/Application Support/spin/plugins/manifests/deploy.json: No such file or directory (os error 2)

$ spin login       
Unknown command: plugin 'login' not found at expected location /Users/radu/Library/Application Support/spin/plugins/manifests/login.json: No such file or directory (os error 2
```

Comment (bacongobbler) [reply src/commands/cloud.rs:11]: Yeah this was part of an earlier experimentation that does not work.

See https://github.com/fermyon/spin/issues/1117#issuecomment-1428908182 for more details.

File: src/commands/cloud.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+use anyhow::Result;
+use clap::Subcommand;
+
+use super::deploy::DeployCommand;
+use super::login::LoginCommand;
+
+/// Commands for publishing applications to Fermyon Cloud.
+#[derive(Subcommand, Debug)]
+pub enum CloudCommands {
+    /// Package and upload an application to Fermyon Cloud.
```
Comment (itowlson) [inline src/commands/cloud.rs:10]: Nit: despite the new naming, this works with the Platform as well - should we say e.g. "to Fermyon" (could be confusing for Fermyon the product vs Fermyon the company) or "to Fermyon Cloud or the Fermyon Platform"?

**Review by itowlson — APPROVED** [review]

File: src/commands.rs [hunk 4]
```diff
@@ -4,11 +4,13 @@
 pub mod bindle;
 /// Commands for building Spin applications.
 pub mod build;
-/// Command for deploying a Spin app to Hippo
+/// Commands for publishing applications to the Fermyon Platform.
```
Comment (calebschoepp) [inline src/commands.rs:7]: Super nit: Throughout you keep referring to the "Fermyon Platform" and I just wanted to make sure that this is our preferred terminology vs "Fermyon Cloud Platform".

Comment (itowlson) [reply src/commands.rs:7]: There are two distinct things, Fermyon Cloud and Fermyon Platform.  These commands work with both.  A new "Fermyon Cloud Platform" term would only add to the confusion I fear!

Comment (bacongobbler) [reply src/commands.rs:7]: I believe the current verbiage is the correct convention.

My understanding is that we use "Fermyon Platform" to refer to the self-hosted as well as the managed service (Fermyon Cloud). We refer to both the self-hosted and the hosted option on [fermyon.dev](https://www.fermyon.dev/).

Comment (calebschoepp) [reply src/commands.rs:7]: Cool, thanks for the thorough response you two 😄 . Looks good.

**Review by michelleN — COMMENTED** [review]
The command help for `spin deploy` and `spin cloud deploy` look different. I think we should make these the same?

```
$ spin deploy --help
spin-deploy
Deploy a Spin application

USAGE:
    spin deploy [OPTIONS]
```

```
$ spin cloud deploy --help
spin-cloud-deploy
Package and upload an application to the Fermyon Platform

USAGE:
    spin cloud deploy [OPTIONS]
```

**Comment by bacongobbler** [comment]
Nice catch. This seems to be due to the fact that `DeployCommand` uses the `clap(about = "")` macro, while `spin cloud deploy` uses the docstring attached to the subcommand.

Fixed in 7b08785.

**Comment by bacongobbler** [comment]
Did the same in 4e62660 for `spin login`.

**Review by michelleN — APPROVED** [review]
Thanks for the changes. This looks great 👍

**Pull request merged**